modules/targets/darwin/options.nix

```diff
@@ -17,6 +17,16 @@ let
       description = "Whether to enable ${name}.";
     };
 
+  isFloat = x:
+    isString x && builtins.match "^[+-]?([0-9]*[.])?[0-9]+$" x != null;
+
+  float = mkOptionType {
```

Comment:
Since there already is a `float` type in Nixpkgs that expects Nix float literals, I think this may be best renamed to something like `floatStr`, similarly maybe `isFloat` above should be `isFloatStr`.

Reply:
Agreed.

Reply:
Actually, I believe that float options should use the `float` type from Nixpkgs. In its current state, the `floatStr` type is going to get serialized as a plain string.

Example:

Here's an configuration that sets an option of the `floatStr` type:

```nix
{ ... }:

{
  targets.darwin.defaults.NSGlobalDomain.NSWindowResizeTime = "0.1";
}
```

The above configuration would generate the following plist for consumption by `defaults import`:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>NSWindowResizeTime</key>
	<string>0.1</string>
</dict>
</plist>
```